File: sdk/javascript/examples/free-interval.ts
import { Gapwise, GapwiseApiError } from "@gapwise/sdk";
const gapwise = new Gapwise();
try {
  const plan = await gapwise.gaps.plan({
    from: "MN",
    to: "IB",
    term: "Fall",
    weekday: "Wednesday",
    startTime: 660,
    endTime: 780,
  });
  console.log(plan.assessment.primary.title, plan.assessment.confidenceLabel);
} catch (error) {
  if (error instanceof GapwiseApiError && error.code === "rate_limited")
    console.error("Retry later", error.requestId);
  else throw error;
}

File: sdk/python/examples/free_interval.py
from gapwise import Gapwise, GapwiseAPIError

with Gapwise() as gapwise:
    try:
        plan = gapwise.gaps.plan(
            from_building="MN",
            to_building="IB",
            term="Fall",
            weekday="Wednesday",
            start_time=660,
            end_time=780,
        )
        print(plan["assessment"]["primary"]["title"])
    except GapwiseAPIError as error:
        if error.code == "rate_limited":
            print("Retry later; request:", error.request_id)
        else:
            raise
